kaspad/src/main.rs

```diff
@@ -17,6 +20,22 @@ pub fn main() {
     let _profiler = dhat::Profiler::builder().file_name("kaspad-heap.json").build();
 
     let args = parse_args();
+
+    match fd_budget::try_set_fd_limit(DESIRED_DAEMON_SOFT_FD_LIMIT) {
+        Ok(limit) => {
+            if limit < MINIMUM_DAEMON_SOFT_FD_LIMIT {
+                println!("Current OS file descriptor limit (soft FD limit) is set to {limit}");
+                println!("The kaspad node requires a setting of at least {DESIRED_DAEMON_SOFT_FD_LIMIT} to operate properly.");
+                println!("Please increase the limits using the following command:");
+                println!("ulimit -n {DESIRED_DAEMON_SOFT_FD_LIMIT}");
```

Comment:
Is it worth making a recommendation in README.md so that this can be done intentionally before the user uses kaspad?

Reply:
Perhaps.  I don't want to touch the readme right now because of another outstanding PR that needs a review.  Now that we have discovered that this works I am much less concerned about this as it appears that rusty will now self-configure in most cases and will only fail to do this in some unusual circumstances, which I am currently not sure what they are... I think the upcoming T11 should show if there are any issues with any types of exotic hardware/setups.